exo/download/shard_download.py

```diff
@@ -27,7 +27,7 @@ def on_progress(self) -> AsyncCallbackSystem[str, Tuple[Shard, RepoProgressEvent
 
 
 class NoopShardDownloader(ShardDownloader):
-  async def ensure_shard(self, shard: Shard) -> Path:
+  async def ensure_shard(self, shard: Shard, inference_engine_name: str) -> Path:
```

Comment:
what's the intention of adding `inference_engine_name` here?

Reply:
Necessary because of the change to the abstractmethod signature

Reply:
Im assuming this is for some future where different inference engines save the model in different ways. Whatever the reason, maybe add it to the docstring for the abstract method `ensure_shard` above?

Reply:
OK... sorry, I see what it's doing now. On first pass I didn't notice that the class in hf_shard_download.py inherits from the parent in this file :)